test/e2e/es/license_test.go

```diff
@@ -49,3 +49,32 @@ func TestEnterpriseLicenseSingle(t *testing.T) {
 		WithSteps(esBuilder.DeletionTestSteps(k)).
 		RunSequential(t)
 }
+
+func TestEnterpriseTrialLicense(t *testing.T) {
+	esBuilder := elasticsearch.NewBuilder("test-es-trial-license").
+		WithESMasterDataNodes(1, elasticsearch.DefaultResources)
+
+	var licenseTestContext elasticsearch.LicenseTestContext
+
+	initStepsFn := func(k *test.K8sClient) test.StepList {
+		return test.StepList{
+			{
+				Name: "Create license test context",
+				Test: func(t *testing.T) {
+					licenseTestContext = elasticsearch.NewLicenseTestContext(k, esBuilder.Elasticsearch)
+				},
+			},
+			licenseTestContext.DeleteEnterpriseTrialLicenseSecret(),
```

Comment:
Should you not delete any license secret in the cluster? Also: could this fail until we implement #2146 and the other license e2e test runs first?

Reply:
> Should you not delete any license secret in the cluster?

Absolutely. I used a different name for the enterprise trial license secret than for the enterprise license secret but in fact it does not help much. I forgot that it's a "global secret".

> Also: could this fail until we implement #2146 and the other license e2e test runs first?

I don't think so. I successfully ran TestEnterpriseLicenseSingle and TestEnterpriseTrialLicense.